pkg/cloudprovider/providers/vsphere/nodemanager.go

```diff
@@ -260,6 +258,101 @@ func (nm *NodeManager) removeNode(node *v1.Node) {
 	nm.nodeInfoLock.Unlock()
 }
 
+func (nm *NodeManager) RegisterVMUUID(nodeName string, vmUUID string) error {
+	nm.vmUUIDConfigMapLock.Lock()
+	defer nm.vmUUIDConfigMapLock.Unlock()
+
+	configMap, err := nm.getVMUUIDConfigMap()
+	if err != nil {
+		glog.Errorf("Error while getting vm uuid configmap. err: %+v", err)
+		return err
+	}
+	if configMap.Data == nil {
+		configMap.Data = map[string]string{}
+	}
+	configMap.Data[nodeName] = vmUUID
+	if err := nm.updateVMUUIDConfigMap(configMap); err != nil {
+		glog.Errorf("Error while registering a new node. nodeName: %s, vmUUID: %s, err: %+v", nodeName, vmUUID, err)
+		return err
+	}
+	return nil
+}
+
+func (nm *NodeManager) GetVMUUID(nodeName string) (string, error) {
+	nm.vmUUIDConfigMapLock.RLock()
+	defer nm.vmUUIDConfigMapLock.RUnlock()
+
+	vmUUID, ok := nm.vmUUIDConfigMapCache[nodeName]
+	if ok {
+		return vmUUID, nil
+	}
+	glog.V(4).Infof("vm uuid for %s not found in cache, checking configmap", nodeName)
+	configMap, err := nm.getVMUUIDConfigMap()
+	if err != nil {
+		glog.Errorf("Error while getting vm uuid configmap. err: %+v", err)
+		return "", err
+	}
+	vmUUID, ok = configMap.Data[nodeName]
+	if !ok {
+		glog.Errorf("vm uuid for %s not found in configmap", nodeName)
+		return "", err
+	}
+	glog.V(4).Infof("vm uuid for %s found in configmap. uuid: %s", nodeName, vmUUID)
+	return vmUUID, nil
+}
+
+func (nm *NodeManager) getVMUUIDConfigMap() (*v1.ConfigMap, error) {
+	if nm.kubeClient == nil {
+		return nil, fmt.Errorf("kubeClient is not initialized yet")
+	}
+	configMap, err := nm.kubeClient.CoreV1().ConfigMaps(nm.vmUUIDConfigMapNamespace).Get(nm.vmUUIDConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		statusErr, ok := err.(*apierrors.StatusError)
+		if ok && statusErr.ErrStatus.Code == http.StatusNotFound {
+			if err := nm.initializeVMUUIDConfigMap(); err != nil {
+				return nil, err
+			}
+			return nm.getVMUUIDConfigMap()
+		}
+		glog.Errorf("Error while getting vm uuid configmap. err: %+v", err)
+		return nil, err
+	}
+	if configMap.Data == nil {
+		nm.vmUUIDConfigMapCache = make(map[string]string)
```

Comment:
Giving configmap read and write permissions to vSphere Cloud Provider after launching k8s cluster will fail the volume operations. This happens due to node events are occurred before giving permissions and nodeName and vmUUID mapping is not cached at that time. Can you fix this?

Reply:
Installing the RBAC permissions via bootstrap policies should fix that right?